### PR TITLE
Use shared error class.

### PR DESCRIPTION
**Pull request by captainbeardo**
Use the Conduit::Error for reporting errors in a normalized way.

### CODE REVIEW

File: lib/conduit/reactor/parsers/base.rb [hunk 1]
```diff
@@ -18,7 +18,7 @@ def initialize(json, http_status = nil)
         @http_status = http_status
         @success = true
       rescue MultiJson::ParseError
-        @json = unexpected_response_hash
+        @json = {errors: {base: "Unexpected response from server."}}
```
Comment (dougperez) [inline lib/conduit/reactor/parsers/base.rb]: Should this be ```unexpected_response```?

Comment (captainbeardo) [inline lib/conduit/reactor/parsers/base.rb]: No, because it will still look in there as if it were the parsed json. The actual error is built later

Comment (dougperez) [inline lib/conduit/reactor/parsers/base.rb]: Ah, right.

**Comment by dougperez** [comment]
👍

**Pull request merged**